ibis/pandas/core.py

```diff
@@ -554,3 +610,113 @@ def canonicalize_context(
             f'begin time {begin} must be before or equal' f' to end time {end}'
         )
     return begin, end
+
+
+class TimeContextRelation(enum.Enum):
```

Comment:
Perhaps move this to a timecontext module

Reply:
Make sense, will move this to an upper level (ibis level) so that we could reuse this module for other backends